codice/ddf
# DDF-3719 Various UI extensibility changes

#### What does this PR do?

- Fixes ace relative module resolution when using symlinks
- Adds remove button to layer-item component if a layer is removable
- Updates map component to listen to mapLayer collection on add/remove
- Updates WMT layer type to use the first available layer if none is provided
  - Warns in the console if not provided
  - Warns in the console if provided but not found
- Prevents removal of user removable layers when syncing with admin imagery providers
- Adds UI as a distribution boot feature

#### Who is reviewing it? 

@mackncheesiest 
@Schachte
@adimka 

#### Select relevant component teams:

@codice/ui 

#### Choose 2 committers to review/merge the PR.

@andrewkfiedler
@bdeining

#### How should this be tested?

- Full build of the ui directory
- Quick build of everything else

To test user removable map layers:
- In the admin console, add two map layers one with the `userRemovable` key and another without
- Open Intrigue and verify that the user can remove one but not the other

To test WMT default layer:
- In the admin console, add [this WMT](https://services.arcgisonline.com/arcgis/rest/services/Demographics/USA_Population_Density/MapServer/WMTS/) layer, but don't specify the `layer` key
- Open Intrigue and verify that the map layer shows up correctly

To prevent regressions:
- Add a few other map layer types and ensure they work as expected
- Perform the previous tests with the cesium

#### What are the relevant tickets?

[DDF-3719](https://codice.atlassian.net/browse/DDF-3719)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
